app/controllers/v0/profile/email_addresses_controller.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module V0
+  module Profile
+    class EmailAddressesController < ApplicationController
+      before_action { authorize :vet360, :access? }
+
+      def create
+        email_address = Vet360::Models::Email.with_defaults(@current_user, email_address_params)
+
+        if email_address.valid?
+          response    = service.post_email email_address
+          transaction = AsyncTransaction::Vet360::EmailTransaction.start(@current_user, response)
+
+          render json: transaction, serializer: AsyncTransaction::BaseSerializer
+        else
+          raise Common::Exceptions::ValidationErrors, email_address
+        end
+      end
+
+      def update
+        email_address = Vet360::Models::Email.with_defaults(@current_user, email_address_params)
+
+        if email_address.valid?
+          response    = service.put_email email_address
+          transaction = AsyncTransaction::Vet360::EmailTransaction.start @current_user, response
+
+          render json: transaction, serializer: AsyncTransaction::BaseSerializer
+        else
+          raise Common::Exceptions::ValidationErrors, email_address
+        end
+      end
+
+      private
+
+      def service
+        Vet360::ContactInformation::Service.new @current_user
```

Comment:
I'm doing something very analogous to this in my branch at the moment, and I have a reminder note to discuss with you guys... I think technically we should be using dependency injection to pass in the service instead of `new` ing it in the class. I'm curious to get y'alls thoughts on this.

Reply:
Yeah, this follows precedence set throughout the other controllers.  I've also seen this to be pretty standard.  Great place to hook in, as we have a handle on the `@current_user` in the controller.